src/git_context.rs

```diff
@@ -0,0 +1,71 @@
+use color_eyre::eyre::{Context, Result};
+use spdx::Expression;
+
+use crate::{
+    cli::FlakeHubPushCli, github::graphql::GithubGraphqlDataResult, revision_info::RevisionInfo,
+};
+
+pub struct GitContext {
+    pub spdx_expression: Option<Expression>,
+    pub repo_topics: Vec<String>,
+    pub revision_info: RevisionInfo,
+}
+
+impl GitContext {
+    pub fn from_cli_and_github(
+        cli: &FlakeHubPushCli,
+        github_graphql_data_result: &GithubGraphqlDataResult,
+    ) -> Result<Self> {
+        // step: validate spdx, backfill from GitHub API
+        let spdx_expression = if cli.spdx_expression.0.is_none() {
+            if let Some(spdx_string) = &github_graphql_data_result.spdx_identifier {
+                tracing::debug!("Recieved SPDX identifier `{}` from GitHub API", spdx_string);
+                let parsed = spdx::Expression::parse(spdx_string)
+                    .wrap_err("Invalid SPDX license identifier reported from the GitHub API, either you are using a non-standard license or GitHub has returned a value that cannot be validated")?;
+                //span.record("spdx_expression", tracing::field::display(&parsed));
```

Comment:
Are we gonna bring this back?

Reply:
I'll personally bring it back if/when we need it.